contract-derive/src/lib.rs

```diff
@@ -87,11 +122,118 @@ pub fn contract(_attr: TokenStream, item: TokenStream) -> TokenStream {
         }
     }).collect();
 
+    let emit_helper = quote! {
+        #[macro_export]
+        macro_rules! get_type_signature {
+            ($arg:expr) => {
+                match stringify!($arg) {
```

Comment:
there's for sure an `alloy` function that does this conversion

Reply:
I believe you can do something like this `(from, to, amount).sol_type_name()` to get signature.
https://docs.rs/alloy-sol-types/latest/alloy_sol_types/trait.SolValue.html